src/Microsoft.SqlTools.ServiceLayer/Profiler/IXEventSession.cs

```diff
@@ -1,35 +1,79 @@
-//
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-//
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+
+using System;
+using System.Threading;
+
+namespace Microsoft.SqlTools.ServiceLayer.Profiler
+{
+    /// <summary>
+    /// Main class for Profiler Service functionality
+    /// </summary>
+    public interface IXEventSession
+    {
+        /// <summary>
+        /// Gets unique XEvent session Id
+        /// </summary>
+        SessionId Id { get; }
+
+        /// <summary>
+        /// Starts XEvent session
+        /// </summary>
+        void Start();
+
+        /// <summary>
+        /// Stops XEvent session
+        /// </summary>
+        void Stop();
+
+        /// <summary>
+        /// Reads XEvent XML from the default session target
+        /// </summary>
+        string GetTargetXml();
+    }
 
-#nullable disable
+    public interface IObservableXEventSession : IXEventSession
+    {
+        IObservable<Contracts.ProfilerEvent> ObservableSessionEvents { get; }
+    }
 
-namespace Microsoft.SqlTools.ServiceLayer.Profiler
-{
-    /// <summary>
-    /// Main class for Profiler Service functionality
+    /// <summary>
+    /// Strong type for use as a dictionary key, simply wraps a string.
+    /// Using a class helps distinguish instances from other strings like viewer id.
     /// </summary>
-    public interface IXEventSession
-    {
-        /// <summary>
-        /// Gets unique XEvent session Id
-        /// </summary>
-        int Id { get; }
-
-        /// <summary>
-        /// Starts XEvent session
-        /// </summary>
-        void Start();
-
-        /// <summary>
-        /// Stops XEvent session
-        /// </summary>
-        void Stop();
-
-        /// <summary>
-        /// Reads XEvent XML from the default session target
-        /// </summary>
-        string GetTargetXml();
+    public class SessionId
+    {
+        private readonly string sessionId;
+        // SQL Server starts session counters at around 64k, so it's unlikely that this process-scoped counter would collide with a real session id
+        // Eventually the profiler extension in ADS will use the string instead of the number.

```

Comment:
How "eventually" is this "eventually"?  Should this change be made now?

Reply:
My intention is to setup a basic workflow first for opening XEL files, and once the framework is there, work on the improvements. 
So, to your question this change should hopefully follow soon.